akka-actor-tests/src/test/scala/akka/pattern/BackoffOnRestartSupervisorSpec.scala

```diff
@@ -186,5 +191,27 @@ class BackoffOnRestartSupervisorSpec extends AkkaSpec with ImplicitSender {
         probe.expectTerminated(supervisor)
       }
     }
+
+    "respect withinTimeRange property of OneForOneStrategy" in {
+      val probe = TestProbe()
+      // The minimum backoff needs to be larger than "withinTimeRange" - otherwise it'll reset
+      val options = Backoff.onFailure(TestActor.props(probe.ref), "someChildName", 500 millis, 10 seconds, 0.0)
+        .withSupervisorStrategy(OneForOneStrategy(withinTimeRange = 300 millis) {
+          case _: TestActor.StoppingException ⇒ SupervisorStrategy.Stop
+        })
+      val supervisor = system.actorOf(BackoffSupervisor.props(options))
+      probe.expectMsg("STARTED")
+      filterException[TestActor.TestException] {
+        probe.watch(supervisor)
+        // Throw something immediately, we should get restarted
+        supervisor ! "THROW"
+        probe.expectMsg("STARTED")
+        // AFter waiting longer than the time range, throw again, we should not be restarted.
+        Thread.sleep(320)
+        supervisor ! "THROW"
+        // Superviser should've terminated.
```

Comment:
Something wrong here, I think.

The `withinTimeRange` should be like a reset of the restart counter, and it only makes sense to use it together with `maxNrOfRetries`.

Example `maxNrOfRetries = 5, withinTimeRange = 10.seconds`.
If you throw 6 times within 10 seconds it will stop.
If you throw 4 times within 10 seconds and then later throw 3 more times it will still restart.

You can think of it as the counter is reset to 0 every 10 seconds.

Look at the tests in https://github.com/akka/akka/blob/master/akka-actor-tests/src/test/scala/akka/actor/RestartStrategySpec.scala